### PR TITLE
Add podUID as a template input in tasklog plugin

### PR DESCRIPTION
**Pull request by goyalankit**
# TL;DR
Add podUID as an additional template input variable so that Log URLs can refer to the UIDs.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
A pod UID is unique across the cluster and logging systems can use them to refer to logs.

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/

## Follow-up issue
_NA_
TODO: update documentation if the changes looks ok

### CODE REVIEW

**Review by hamersaw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Comment by hamersaw** [comment]
@goyalankit thanks for this! Mind submitting a small PR to update the docs as well? Should be [this page](https://docs.flyte.org/projects/cookbook/en/latest/auto/deployment/configure_logging_links.html#sphx-glr-auto-deployment-configure-logging-links-py) that is located [here](https://github.com/flyteorg/flytesnacks/blob/master/cookbook/deployment/configure_logging_links.py).
